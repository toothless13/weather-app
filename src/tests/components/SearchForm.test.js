import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm component", () => {
  const validProps = {
    searchText: "Oxford",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a button with the correct text", () => {
    const { getByText, getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />,
    );

    expect(getByRole("button")).toBeInTheDocument();
    expect(getByRole("button")).toHaveTextContent("Search");
    expect(getByText("Search")).toBeTruthy();
  });
});
