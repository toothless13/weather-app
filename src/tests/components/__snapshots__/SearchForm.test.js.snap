// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value="Oxford"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
